[AOTInductor] Add tensor_constantX to pass constant buffer update's check

muchulee8 opened the pull request:
Summary:
During tracing, some constants (tensor_constant{idx}) are being generated internally.
Those constants are neither parameters or buffers, and users have zero control on them.

To accomodate this, we should allow users not passing in those constants generated internally but still be able the constants in the model.

Test Plan:
Included in commit.
```
build/bin/test_aot_inductor
```

Differential Revision: D55286634

chenyang78 reviewed (approved):
LGTM. Thanks!

khabinov commented on torch/csrc/inductor/aoti_runtime/model_container.h:
```diff
@@ -218,6 +218,9 @@ class AOTInductorModelContainer {
     pending_models_available_.notify_one();
   }
 
+  bool _is_tensor_constant(const std::string& constant_name) {
```
Nit: can mark this method as `const`, e.g.:
```
bool _is_tensor_constant(const std::string& constant_name) const {
```

khabinov commented on torch/csrc/inductor/aoti_runtime/model_container.h:
```diff
@@ -243,6 +247,13 @@ class AOTInductorModelContainer {
         auto it = constants_map.find(constant_name);
         if (it == constants_map.end()) {
+          if (_is_tensor_constant(constant_name)) {
+            // tracing sometimes create tensors that are non-existent in
```
Nit: create -> creates

khabinov commented on torch/csrc/inductor/aoti_runtime/model_container.h:
```diff
@@ -243,6 +247,13 @@ class AOTInductorModelContainer {
         if (it == constants_map.end()) {
+          if (_is_tensor_constant(constant_name)) {
+            // tracing sometimes create tensors that are non-existent in
+            // original merge net. We could skip those and do a direct copy.
```
Merge? Can we remove it?

muchulee8 committed:
[AOTInductor] Add tensor_constantX to pass constant buffer update's check (#122562)

Summary:

During tracing, some constants (tensor_constant{idx}) are being generated internally.
Those constants are neither parameters or buffers, and users have zero control on them.

To accomodate this, we should allow users not passing in those constants generated internally but still be able the constants in the model.

Test Plan:
Included in commit.
```
build/bin/test_aot_inductor
```

Reviewed By: Indicator, khabinov, chenyang78, aakhundov

Differential Revision: D55286634
